### PR TITLE
Fixes #4195 : Added dark mode support to QuestionPlayerActivity and ExplorationActivity

### LINKED ISSUE #4195: Add dark mode support to QuestionPlayerActivity and ExplorationActivity
Add dark mode support to QuestionPlayerActivity and ExplorationActivity (includes state fragment).

Go through this [wiki](https://github.com/oppia/oppia-android/wiki/Dark-Mode) to get started.

### PR DESCRIPTION
**Pull request by bhaktideshmukh**
## Explanation

Fixes #4195 .
Added dark mode support to QuestionPlayerActivity and ExplorationActivity.

Mocks Links

- QuestionPlayer : https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/8fbb663a-39aa-4be1-83bd-73bde6f85dcc/specs/
- ExplorationActivity : https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/40d6cf5f-f673-42c8-af15-035f711ef6b9/specs/

**ScreenShots**

**_QuestionPlayerActivity_**

| Before  | After dark mode|
|-------|---|
|||

| Before  | After dark mode |
|-------|---|
|||

| Before  | After dark mode |
|-------|---|
|||

**_ExplorationActivity_**

| Before  | After dark mode |
|-------|---|
|||

| Before  | After dark mode |
|-------|---|
|||

| Before  | After dark mode |
|-------|---|
|||

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

**Comment by rt4914** [comment]
Un-assigning until approved / re-assigned by @ayush0402

**Comment by bhaktideshmukh** [comment]
PTAL @ayush0402 
Thanks

File: app/src/main/res/layout-land/question_player_fragment.xml
Comment (ayush0402) [inline app/src/main/res/layout-land/question_player_fragment.xml]: Please update all colors to follow newer convention having file name in the color declaration.

File: app/src/main/res/values/component_colors.xml
Comment (ayush0402) [inline app/src/main/res/values/component_colors.xml]: `<!-- Question Player Activity -->`

File: app/src/main/res/values/component_colors.xml
Comment (ayush0402) [inline app/src/main/res/values/component_colors.xml]: `<!-- Exploration Activity -->`

File: app/src/main/res/layout/question_player_activity.xml
Comment (ayush0402) [inline app/src/main/res/layout/question_player_activity.xml]: This should be a color from component_colors.xml

**Review by ayush0402 — CHANGES_REQUESTED** [review]
The approach seems to be good. Took a pass and left some comments. Please update the name convention and assign me back. Rest seems to be fine might need to rename some colors later.

File: app/src/main/res/layout-land/question_player_fragment.xml
Comment (bhaktideshmukh) [reply app/src/main/res/layout-land/question_player_fragment.xml]: > Please update all colors to follow newer convention having file name in the color declaration.

Can you please elaborate how should I update the name here.
I have already added the file name in the beginning of the color name.

File: app/src/main/res/values/component_colors.xml
Comment (bhaktideshmukh) [reply app/src/main/res/values/component_colors.xml]: Done

File: app/src/main/res/values/component_colors.xml
Comment (bhaktideshmukh) [reply app/src/main/res/values/component_colors.xml]: Done

File: app/src/main/res/layout/question_player_activity.xml
Comment (bhaktideshmukh) [reply app/src/main/res/layout/question_player_activity.xml]: Done

**Comment by rt4914** [comment]
@ayush0402 Can you review it once again as you have suggested changes.

File: app/src/main/res/layout-land/question_player_fragment.xml
Comment (rt4914) [inline app/src/main/res/layout-land/question_player_fragment.xml]: For this create a new color `question_player_fragment_background_color` as this is a fragment file and not an activity file. Also, removed `layout` word as this does not add any extra info.

File: app/src/main/res/layout-land/question_player_fragment.xml
Comment (rt4914) [inline app/src/main/res/layout-land/question_player_fragment.xml]: Same -> `question_player_fragment_...`

File: app/src/main/res/layout-land/question_player_fragment.xml
Comment (rt4914) [inline app/src/main/res/layout-land/question_player_fragment.xml]: Ditto

File: app/src/main/res/layout/content_item.xml
Comment (rt4914) [inline app/src/main/res/layout/content_item.xml]: Considering this color might be getting used at a lot of places, I suggest you name it as follows:

`shared_player_text_view_primary_color`

Comment (rt4914) [reply app/src/main/res/layout/content_item.xml]: Or if you want to create specific colors then name it as `content_item_text_view_color`

File: app/src/main/res/layout/exploration_fragment.xml
Comment (rt4914) [inline app/src/main/res/layout/exploration_fragment.xml]: Replace -> `activity` -> `fragment`


Also if all background colors are same then you can also create a shared one like `shared_player_background_color`

File: app/src/main/res/layout/feedback_item.xml
Comment (rt4914) [inline app/src/main/res/layout/feedback_item.xml]: This naming is incorrect because this file is getting used in exploration player too.

Name something like 

`feedback_item_text_view_primary_color`

Comment (rt4914) [reply app/src/main/res/layout/feedback_item.xml]: Ditto for other files too.

File: app/src/main/res/layout/previous_responses_header_item.xml
Comment (rt4914) [inline app/src/main/res/layout/previous_responses_header_item.xml]: Ditto for this file

File: app/src/main/res/layout/question_player_fragment.xml
Comment (rt4914) [inline app/src/main/res/layout/question_player_fragment.xml]: `activity` -> `fragment` replace it.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@bhaktideshmukh Suggested changes. Thanks.

**Comment by rt4914** [comment]
@bhaktideshmukh I think all colors should have `component_color...` keyword ahead of it but I am unable see that rule in wiki https://github.com/oppia/oppia-android/wiki/Dark-Mode

Will confirm this with Ayush.


@bhaktideshmukh DON'T apply the changes that I have suggested right now as I have some confusion with the rules and therefore I think let's first clarify them. Also at a lot of places I have suggested `fragment` instead of `activity` which might not be correct as per wiki.

File: app/src/main/res/layout-land/question_player_fragment.xml
Comment (bhaktideshmukh) [reply app/src/main/res/layout-land/question_player_fragment.xml]: Done

File: app/src/main/res/layout-land/question_player_fragment.xml
Comment (bhaktideshmukh) [reply app/src/main/res/layout-land/question_player_fragment.xml]: Done

File: app/src/main/res/layout/content_item.xml
Comment (bhaktideshmukh) [reply app/src/main/res/layout/content_item.xml]: To me first one looks much better

File: app/src/main/res/layout/exploration_fragment.xml
Comment (bhaktideshmukh) [reply app/src/main/res/layout/exploration_fragment.xml]: Done

File: app/src/main/res/layout/question_player_fragment.xml
Comment (bhaktideshmukh) [reply app/src/main/res/layout/question_player_fragment.xml]: Done

**Review by rt4914 — CHANGES_REQUESTED** [review]
@bhaktideshmukh Please add dark-mode mock links in PR description and also apply the suggested changes.

**Comment by bhaktideshmukh** [comment]
> @bhaktideshmukh Please add dark-mode mock links in PR description and also apply the suggested changes.

I have added some screenshots in PR description. Do I have to add the screenshots of all the pages for dark mode of QuestionPlayerActivity and ExplorationActivity in a doc and then attach the link here?
And about suggested changes I have already added all the suggested changes.

File: app/src/main/res/layout-land/question_player_fragment.xml
Comment (rt4914) [inline app/src/main/res/layout-land/question_player_fragment.xml]: All colors should follow this format:

`component_color_<activity-name>_<color-usage/type>_color`

For this it should be `component_color_question_player_activity_background_color`

File: app/src/main/res/values/color_palette.xml
Comment (rt4914) [inline app/src/main/res/values/color_palette.xml]: Remove this blank line

**Review by rt4914 — CHANGES_REQUESTED** [review]
Now sure why in last review the comments were not shown but basically follow the convention mentioned in the below comment.

File: app/src/main/res/layout-land/question_player_fragment.xml
Comment (bhaktideshmukh) [reply app/src/main/res/layout-land/question_player_fragment.xml]: Got it

**Review by Akshatkamboj14 — CHANGES_REQUESTED** [review]
1. I think exploration activity's card background is not correct , it should be **#32363B** .
2. Question Player Activity's toolbar color should be changed too, in the Dark Mode mocks its mentioned **#035D63**.
3. Kindly PTAL at the buttons, it should also have borders, like this - 
image
image
and background color of the continue button have changes too.

4. the progress bar color also have changes left, it should be like this
image

**Comment by rt4914** [comment]
For now we have finalised a different approach, which is we will implement dark-mode for items like content, feedback, interactions, buttons, first and then in the end focus on fragment + activity.

Therefore this PR is on hold for now.

**Comment by bhaktideshmukh** [comment]
As per decided will be doing this PR in subparts starting from PR no. https://github.com/oppia/oppia-android/pull/4377 hence closing this one.

**Pull request closed without merging**
